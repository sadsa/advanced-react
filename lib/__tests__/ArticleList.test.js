import React from 'react';
import ArticleList from '../components/ArticleList';
import { data } from '../testData';
import ReactTestRenderer from 'react-test-renderer';

const testProps = {
  articles: [
    { id: 'a' },
    { id: 'b' }
  ],
  actions: {
    lookupAuthorById: jest.fn(() => data.authors[0] )
  }
};

describe('Article List Component', () => {
  it('should render correctly', () => {
    const domTree = ReactTestRenderer.create(
      <ArticleList {...testProps} />
    ).toJSON();

    expect(domTree).toMatchSnapshot();
    expect(domTree.children.length).toBeGreaterThan(0);
  });
});