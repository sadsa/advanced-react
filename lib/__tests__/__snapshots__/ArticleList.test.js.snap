// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Article List Component should render correctly 1`] = `
<div>
  <div>
    <div />
    <div>
      Invalid Date
    </div>
    <div>
      <a
        href="https://twitter.com/acdlite"
      >
        Andrew
         
        Clark
      </a>
    </div>
    <div />
  </div>
  <div>
    <div />
    <div>
      Invalid Date
    </div>
    <div>
      <a
        href="https://twitter.com/acdlite"
      >
        Andrew
         
        Clark
      </a>
    </div>
    <div />
  </div>
</div>
`;
